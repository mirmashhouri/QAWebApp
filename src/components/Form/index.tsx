import { memo } from "react";
import { useForm } from "react-hook-form";
import { Question } from "../../services/questionService";
import styles from "./styles.module.scss";

// interface Props {
//   item: Question;
// }

const Form = ({ onAdd }: any): JSX.Element => {
  // const [isVisible, setIsVisible] = useState(false);
  // const QuestionOnClick = () => {
  //   setIsVisible(!isVisible);
  // };
  let lastId = 2;
  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm();
  const onSubmit = (data: any) => {
    lastId += 1;
    data.id = lastId;
    onAdd(data as Question);
  };
  console.log(errors);

  return (
    <form className={styles.item} onSubmit={handleSubmit(onSubmit)}>
      <ul>
        <li>
          <input
            type="text"
            placeholder="question"
            {...register("question", { required: true, maxLength: 198 })}
          />
        </li>
        <li>
          <textarea
            {...register("answer", { required: true, maxLength: 2000 })}
          />
        </li>
        <li>
          <input type="submit" />
        </li>
      </ul>
    </form>
  );
};

export default memo(Form);
