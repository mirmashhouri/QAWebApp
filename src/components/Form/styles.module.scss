@import "../../theme/variables";

.form {
  border-bottom: 1px solid $color-white;
  margin: 10px 0;
  padding: 5px;
}
.error {
  color: #dc3545;
  font-size: 0.875em;
  margin-top: 0.25rem;
  width: 100%;
}
