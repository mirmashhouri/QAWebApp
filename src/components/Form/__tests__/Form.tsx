import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

import Form from "..";

describe("<Form />", () => {
  it("renders", () => {
    const tree = render(
      <MemoryRouter>
        <Form />
      </MemoryRouter>
    ).container.firstChild;

    expect(tree).toMatchSnapshot();
  });
});
