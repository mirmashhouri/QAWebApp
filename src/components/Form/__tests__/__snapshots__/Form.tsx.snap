// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> renders 1`] = `
<form
  class="item"
>
  <ul>
    <li>
      <input
        name="question"
        placeholder="question"
        type="text"
      />
    </li>
    <li>
      <textarea
        name="answer"
      />
    </li>
    <li>
      <input
        type="submit"
      />
    </li>
  </ul>
</form>
`;
